type-definitions/immutable.js.flow

```diff
@@ -526,8 +526,14 @@ declare module 'immutable' {
     updateIn(keyPath: ESIterable<any>, notSetValue: any, value: any): Map<K,V>;
     updateIn(keyPath: ESIterable<any>, value: any): Map<K,V>;
 
-    mergeIn(keyPath: ESIterable<any>, ...iterables: ESIterable<any>[]): Map<K,V>;
-    mergeDeepIn(keyPath: ESIterable<any>, ...iterables: ESIterable<any>[]): Map<K,V>;
+    mergeIn(
+      keyPath: ESIterable<any>,
+      ...iterables: (ESIterable<any> | { [key: string]: any })[]
+    ): Map<K,V>;
```

Comment:
I used `any` b/c of this: http://facebook.github.io/immutable-js/docs/#/Map/merge

```
If any of the values provided to merge are not Iterable (would return false for Immutable.Iterable.isIterable) then they are deeply converted via Immutable.fromJS before being merged. However, if the value is an Iterable but includes non-iterable JS objects or arrays, those nested values will be preserved.
```

So if `V` is also `Map` or `List`, I can pass `Object` or `Array` as `V` and it still should be valid as it will be converted to `Map` or `List` by `fromJS`.